test/unit/abilities/provider_any_test.rb

```diff
@@ -71,57 +81,112 @@ def test_policies_not_tenant
+    end
 
+    test "member has :partners can't show AccountRelatedEvent when there is no service" do
+      user.member_permission_service_ids = nil # All services allowed
       user.member_permission_ids = [:partners]
-      assert_can ability, :show, Accounts::AccountCreatedEvent.create(account, user)
+      assert_cannot ability, :show, Accounts::AccountCreatedEvent.create(@buyer_no_service, @buyer_no_service_user)
     end
 
-    class ShowAlertRelatedEventTest < ProviderAnyTest
-      setup do
-        service = FactoryBot.create(:simple_service, account: @account)
-        plan = FactoryBot.create(:simple_application_plan, issuer: service)
-        cinstance = FactoryBot.create(:cinstance, plan: plan)
-        alert = FactoryBot.create(:limit_violation, cinstance: cinstance)
-        @limit_violation_reached_provider_event = Alerts::LimitViolationReachedProviderEvent.create(alert)
-      end
+    test 'admin can show AccountRelatedEvent when there is no service' do
+      admin = account.first_admin
+      ability = Ability.new(admin)
+      assert_can ability, :show, Accounts::AccountCreatedEvent.create(@buyer_no_service, @buyer_no_service_user)
+    end
 
-      attr_reader :limit_violation_reached_provider_event
+    test 'member has :partners and access to all services can show AccountRelatedEvent if there is a service' do
+      user.member_permission_service_ids = nil
+      user.member_permission_ids = [:partners]
 
-      test 'cannot show AlertRelatedEvent when user does not have :monitoring' do
-        assert_not user.has_permission? :monitoring
-        assert_cannot ability, :show, limit_violation_reached_provider_event
-      end
+      assert_can ability, :show, Accounts::AccountCreatedEvent.create(@buyer, @buyer_user)
+    end
 
-      # If the user has service-related permission, and no :services permission, it means it has access to all services
-      test 'can show AlertRelatedEvent when user has :monitoring and the user has access to all services' do
-        user.member_permission_ids = [:monitoring]
-        assert user.has_permission? :monitoring
+    test 'member has :partners and access to one service can show AccountRelatedEvent if there is that service' do
+      user.member_permission_service_ids = [@service.id]
+      user.member_permission_ids = [:partners]
 
-        assert_can ability, :show, limit_violation_reached_provider_event
-      end
+      assert_can ability, :show, Accounts::AccountCreatedEvent.create(@buyer, @buyer_user)
+    end
 
-      test 'cannot show AlertRelatedEvent when user has :monitoring and does not have access to the service' do
-        user.member_permission_ids = [:monitoring]
-        user.member_permission_service_ids = []
+    test "member has :partners and access to a service can't show AccountRelatedEvent if there is another service" do
+      service = FactoryBot.create(:simple_service, account: @account)
```

Comment:
a bit confusing IMO to use `service` to refer to a different service. 
If we add and some point an `attr_reader :service`, this would confuse things.

I'd suggest calling it `another_service` or similar.

Reply:
I applied the suggestion. Also reworded some test descriptions to be (hopefully) easier to understand. https://github.com/3scale/porta/pull/4029/commits/6d0f18ae635e23859fe83e1b771ccd266f4fa0e4